XenModel/XenAPI/HTTP.cs

```diff
@@ -299,6 +357,39 @@ public static Uri BuildUri(string hostname, string path, params object[] args)
             return uri.Uri;
         }
 
+        private static List<string> GetFields(List<string> lines, string field)
+        {
+            List<string> fields = new List<string>();
+            foreach (string line in lines)
+                if (line.StartsWith(field))
+                    fields.Add(line);
+            return fields;
+        }
+
+        private static string GetField(List<string> lines, string field)
+        {
+            foreach (string line in lines)
+                if (line.StartsWith(field))
+                    return line;
+            return null;
+        }
+
+        private static string GetPartOrNull(string str, int partIndex)
+        {
+            string[] parts = str.Split(new char[] { ' ' }, partIndex + 2, StringSplitOptions.RemoveEmptyEntries);
+            return partIndex < parts.Length - 1 ? parts[partIndex] : null;
+        }
+
+        private static void RemoveNewLineEnding(ref string str)

```

Comment:
You can use str.TrimEnd('\r', '\n'), in which case you don't need a new method either, you can call it inline.